File: go/problem_122_best_time_to_buy_ii.py
# https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/?

class Solution:
    def maxProfit(self, prices: List[int]) -> int:
        profit = 0

        for i in  range(1, len(prices)):
            result = prices[i] - prices[i-1]

            if result > 0:
                profit += result
        
        return profit
        

File: java/Problem122_BestTimeToBuyII.java
// https://leetcode.com/problems/best-time-to-buy-and-sell-stock-ii/

class Solution {
    public int maxProfit(int[] prices) {
        int profit = 0;

        for (int i=1; i<prices.length; i++) {
            int result = prices[i]-prices[i-1];

            if (result > 0) {
                profit += result;
            }
        }

        return profit;
    }
}
